eng/common/pipelines/templates/steps/cosmos-emulator.yml

```diff
@@ -9,8 +9,92 @@ steps:
+    displayName: Delete Cosmos DB Emulator data
+  - powershell: |
+      Write-Host "Getting Cosmos DB Emulator Version"
+      $ProductName = "Azure Cosmos DB Emulator"
+      $Emulator = (Join-Path $env:temp (Join-Path $ProductName "CosmosDB.Emulator.exe"))
+      $fileVersion = Get-ChildItem $Emulator
+      Write-Host $Emulator $fileVersion.VersionInfo
+    displayName: Get Cosmos DB Emulator Version
   - powershell: |
       Write-Host "Launching Cosmos DB Emulator"
-      Import-Module "$env:temp\Azure Cosmos DB Emulator\PSModules\Microsoft.Azure.CosmosDB.Emulator"
-      Start-CosmosDbEmulator -NoUI ${{ parameters.StartParameters }}
+      $ProductName = "Azure Cosmos DB Emulator"
+      $Emulator = (Join-Path $env:temp (Join-Path $ProductName "CosmosDB.Emulator.exe"))
+      if (!(Test-Path $Emulator)) {
+        Write-Error "The emulator is not installed where expected at '$Emulator'"
+        return
+      }
+
+      $process = Start-Process $Emulator -ArgumentList "/getstatus" -PassThru -Wait
+
+      switch ($process.ExitCode) {
+        1 {
+          Write-Host "The emulator is already starting"
+          return
+        }
+        2 {
+          Write-Host "The emulator is already running"
+          return
+        }
+        3 {
+          Write-Host "The emulator is stopped"
+        }
+        default {
+          Write-Host "Unrecognized exit code $process.ExitCode"
+          return
+        }
+      }
+
+      $argumentList = ""
+      if (-not [string]::IsNullOrEmpty("${{ parameters.StartParameters }}")) {
+          $argumentList += , "${{ parameters.StartParameters }}"
+      }
+
+      Write-Host "Starting emulator process: $Emulator $argumentList"
+      $process=Start-Process $Emulator -ArgumentList $argumentList -ErrorAction Stop -PassThru
+      Write-Host "Emulator process started: $($process.Name), $($process.FileVersion)"
+      $Timeout = 600
+      $result="NotYetStarted"
+      $complete = if ($Timeout -gt 0) {
+        $start = [DateTimeOffset]::Now
+        $stop = $start.AddSeconds($Timeout)
+        {
+          $result -eq "Running" -or [DateTimeOffset]::Now -ge $stop
+        }
+      }
+      else {
+        {
+          $result -eq "Running"
+        }
+      }
+
+      do {
+        $process = Start-Process $Emulator -ArgumentList "/getstatus" -PassThru -Wait
+
+        switch ($process.ExitCode) {
+          1 {
+            Write-Host "The emulator is starting"
+          }
+          2 {
+            Write-Host "The emulator is running"
+            $result="Running"
+            return
+          }
+          3 {
+            Write-Host "The emulator is stopped"
+          }
+          default {
+            Write-Host "Unrecognized exit code $process.ExitCode"
+          }
+        }
+
+        Start-Sleep -Seconds 5
+      }
+      until ($complete.Invoke())
+
+      Write-Error "The emulator failed to reach Running status within ${Timeout} seconds"
```

Comment:
Agreed - it would be good to allow specifying the actual location in the Start-CosmosEmulator etc. powershell cmdlets instead of always preferring the locally installed one and only trying to use the local one when none is installed system-wide.

Reply:
@moderakh are you also referring to the emulator's powershell support for getting the emulator's exe?

Reply:
@milismsft yes.

Reply:
Yep, I'll do this with the next Cosmos public emulator update.